federation/pkg/dnsprovider/providers/coredns/coredns.go

```diff
@@ -63,16 +97,24 @@ func newCoreDNSProviderInterface(config io.Reader) (*Interface, error) {
 		}
 		etcdEndpoints = cfg.Global.EtcdEndpoints
 		dnsZones = cfg.Global.DNSZones
+		certFile = cfg.Global.CertFile
+		keyFile = cfg.Global.KeyFile
+		caFile = cfg.Global.CAFile
 	}
 	glog.Infof("Using CoreDNS DNS provider")
 
 	if dnsZones == "" {
 		return nil, fmt.Errorf("Need to provide at least one DNS Zone")
 	}
 
+	etcdTransport, err := newTransportForETCD2(certFile, keyFile, caFile)
```

Comment:
`newTransportForETCD2` will only succeed in creating an HTTPS transport- there should still be an option to use default HTTP transport as well (perhaps if certs paths are all blank?)